registry/remote/example_test.go

```diff
@@ -0,0 +1,371 @@
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+	// **You still need to validate the digest**
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) {
+		panic(err) // Handle error
+	}
+
+	// Output:
+	// Example blob content
+}
+
+// ExampleRepository_Fetch_byDigest gives example snippets for downloading a blob by digest.
+func ExampleRepository_Fetch_byDigest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+	descriptor, err := repo.Resolve(ctx, digest) // We still need to resolve first, don't create a new descriptor with the digest, blob size is unknown
+	if err != nil {
+		panic(err) // Handle error
+	}
+	r, err := repo.Fetch(ctx, descriptor) // Fetch the blob from the repository
+	if err != nil {
+		panic(err) // Handle error
+	}
+	defer r.Close() // don't forget to close
+
+	// You can 1) fetch a whole blob
+	pulledBlob, err := io.ReadAll(r)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println(string(pulledBlob))
+	// Fetch() doesn't provide validation, you need to do it **
+	if descriptor.Digest != ocidigest.FromBytes(pulledBlob) {
+		panic(err) // Handle error
+	}
+
+	// or 2) do partially read, if the remote registry supports
+	if seeker, ok := r.(io.ReadSeeker); ok {
+		_, err = seeker.Seek(8, io.SeekStart)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		pulledBlob, err := io.ReadAll(r)
+		if err != nil {
+			panic(err) // Handle error
+		}
+		fmt.Println(string(pulledBlob))
+	}
+
+	// Output:
+	// Example blob content
+	// blob content
+}
+
+// ExampleRepository_Tag gives example snippets for tagging a descriptor.
+func ExampleRepository_Tag() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName) // Get the repository from registry
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// suppose we are going to tag a blob with below digest
+	digest := "sha256:aafc6b9fa2094cbfb97eca0355105b9e8f5dfa1a4b3dbe9375a30b836f6db5ec"
+
+	// 1. Resolve the target desc
+	descriptor, err := repo.Resolve(ctx, digest)
+	if err != nil {
+		panic(err) // Handle error
+	}
+
+	// 2. Tag the resolved desc
+	tag := "latest"
+	err = repo.Tag(ctx, descriptor, tag)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	fmt.Println("Succeed")
+
+	// Output:
+	// Succeed
+}
+
+// ExampleRegistry_Repositories gives example snippets for listing respositories in a HTTPS registry with pagination.
+func ExampleRegistry_Repositories() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err) // Handle error
+	}
+	// If you want to play with your local registry, try to override
```

Comment:
nit: Try using objective wording and try to avoid words like _you_, _he_, _she_, _I_, _we_.